pkg/storage/bloom/v1/block.go

```diff
@@ -176,6 +176,6 @@ outer:
 		// chunk passed all searches, add to the list of chunks to download
 		mustCheck = append(mustCheck, chk)
 
-	}
+	} // again is this desired behaviour. only returns must check, not in bloom?
```

Comment:
Yes. A positive test from a bloom filter doesn't tell for certain if a key was indexed, but a negative test is certain a key wasn't indexed. 

We send a list of chunks to the blooms. The ones that don't pass the queries are ignored (they're guaranteed to not have the data we want), and we return the list of chunks we'll need to scan.